tests/protocol/test_extensions_goal_state.py

```diff
@@ -5,45 +5,65 @@
 
 from azurelinuxagent.common.protocol.extensions_goal_state import ExtensionsGoalState, GoalStateMismatchError
 from azurelinuxagent.common.utils.flexible_version import FlexibleVersion
+from azurelinuxagent.common.utils import textutil
 from tests.protocol.mocks import mockwiredata, mock_wire_protocol
 from tests.tools import AgentTestCase, PropertyMock, patch
 
 
 class ExtensionsGoalStateTestCase(AgentTestCase):
+    def test_compare_should_succeed_when_extensions_config_and_vm_settings_are_equal(self):
+        with mock_wire_protocol(mockwiredata.DATA_FILE_VM_SETTINGS) as protocol:
+            from_extensions_config = protocol.client.get_extensions_goal_state()
+            from_vm_settings = protocol.client._extensions_goal_state_from_vm_settings
+
+            try:
+                ExtensionsGoalState.compare(from_extensions_config, from_vm_settings)
+            except Exception as exception:
+                self.fail("Compare goal state failed: {0}".format(textutil.format_exception(exception)))
+
     def test_compare_should_report_mismatches_between_extensions_config_and_vm_settings(self):
         with mock_wire_protocol(mockwiredata.DATA_FILE_VM_SETTINGS) as protocol:
             from_extensions_config = protocol.client.get_extensions_goal_state()
             from_vm_settings = protocol.client._extensions_goal_state_from_vm_settings
 
-            def test_property(name, value):
-                full_name = "azurelinuxagent.common.protocol.extensions_goal_state_from_vm_settings.ExtensionsGoalStateFromVmSettings.{0}".format(name)
-                with patch(full_name, new_callable=PropertyMock) as property_mock:
-                    property_mock.return_value = value
-                    with self.assertRaisesRegexCM(GoalStateMismatchError, r"Attribute: {0}.*{1}".format(name, re.escape(str(value))), re.DOTALL):
-                        ExtensionsGoalState.compare(from_extensions_config, from_vm_settings)
-
-            test_property("activity_id",             'MOCK_ACTIVITY_ID')
-            test_property("correlation_id",          'MOCK_CORRELATION_ID')
-            test_property("created_on_timestamp",    'MOCK_TIMESTAMP')
-            test_property("status_upload_blob",      'MOCK_UPLOAD_BLOB')
-            test_property("status_upload_blob_type", 'MOCK_UPLOAD_BLOB_TYPE')
-            test_property("required_features",       ['MOCK_REQUIRED_FEATURE'])
-            test_property("on_hold",                 False)
-
-            agent_manifests = copy.deepcopy(from_vm_settings.agent_manifests)
-            agent_manifests[0].family = 'MOCK_FAMILY'
-            test_property("agent_manifests", agent_manifests)
-
-            extensions = copy.copy(from_vm_settings.extensions)
-            extensions[0].name = 'MOCK_NAME'
-            test_property("extensions", extensions)
+            def assert_compare_raises(setup_copy, failing_attribute):
+                from_vm_settings_copy = copy.deepcopy(from_vm_settings)
```

Comment:
@larohra : this now implements the deepcopy you pointed out in #2430.

Again, thanks for pointing that out!